### PR TITLE
calibre-web: fix build

### LINKED ISSUE #121717: Calibre-web fails to build on unstable
## Issue description

I just updated my flakes input, and noticed that `calibre-web` now fails to build.

Here's an extract from the build log https://pastebin.com/XpyV3w5t.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.30, NixOS, 21.05.20210426.8e4fe32 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210326_dd77f71`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

The actual revision used by my flakes: 7cb76200088f45cd24a9aa67fd2f9657943d78a4

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
fixes https://github.com/NixOS/nixpkgs/issues/121717

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/calibre-web/default.nix [hunk 1]
```diff
@@ -18,7 +18,9 @@ python3.pkgs.buildPythonApplication rec {
 
   prePatch = ''
     substituteInPlace setup.cfg \
+        --replace "singledispatch>=3.4.0.0,<3.5.0.0" "" \
```
Comment (pborzenkov) [inline pkgs/servers/calibre-web/default.nix:21]: Why did you remove this version constraint completely?

Comment (dotlambda) [reply pkgs/servers/calibre-web/default.nix:21]: > Backport functools.singledispatch from Python 3.4 to Python 2.6-3.3.

https://pypi.org/project/singledispatch/

Comment (dotlambda) [reply pkgs/servers/calibre-web/default.nix:21]: But actually, https://github.com/janeczku/calibre-web/pull/1977

**Review by pborzenkov — APPROVED** [review]

**Review by ambroisie — APPROVED** [review]
LGTM
